### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by Idclip**
Caught with assertions enabled in msvc CRT

### CODE REVIEW

File: openvdb/openvdb/tools/VolumeToMesh.h [hunk 1]
```diff
@@ -5085,7 +5085,7 @@ VolumeToMesh::operator()(const InputGridType& inputGrid)
         if (referenceMeshing) {
             mPointFlags.resize(mPointListSize);
             op.setRefData(*refInputTree, *refPointIndexTree, *refSignFlagsTree,
-                mQuantizedSeamPoints.get(), &mPointFlags.front());
+                mQuantizedSeamPoints.get(), mPointFlags.data());
```
Comment (Idclip) [inline openvdb/openvdb/tools/VolumeToMesh.h:5088]: This was the only failure, the above id drive-by

**Comment by Idclip** [comment]
https://github.com/actions/virtual-environments/issues/2667

:(

File: openvdb/openvdb/tools/VolumeToMesh.h [hunk 2]
```diff
@@ -3757,7 +3757,7 @@ ComputeAuxiliaryData<InputTreeType>::ComputeAuxiliaryData(
     Index32TreeType& pointIndexTree,
     InputValueType iso)
     : mInputAccessor(inputTree)
-    , mIntersectionNodes(&intersectionLeafNodes.front())
```
Comment (jmlait) [inline openvdb/openvdb/tools/VolumeToMesh.h:3760]: This also looks like a failure?

File: openvdb/openvdb/unittest/TestStreamCompression.cc [hunk 3]
```diff
@@ -243,7 +242,7 @@ TEST_F(TestStreamCompression, testBlosc)
         // compress
 
         std::vector<int> smallBuffer;
-        smallBuffer.reserve(count);
```
Comment (jmlait) [inline openvdb/openvdb/unittest/TestStreamCompression.cc:246]: And another correct looking substantial change.

**Review by jmlait — APPROVED** [review]

**Pull request merged**